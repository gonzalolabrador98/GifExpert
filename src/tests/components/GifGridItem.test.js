import { GifGridItem} from '../../components/GifGridItem';
import { shallow } from 'enzyme';
import React from 'react';
import { getByDisplayValue } from '@testing-library/react';


describe('Pruebas en GifGridItem', () => {

    const title='Un titulo'
    const url= 'https://elquetepincha.com'
    const wrapper = shallow( <GifGridItem title={title} url={url}/> )

    test('debe mostrar las pruebas correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();
    })

    test('debe mostrar el parrafo', () => {
        
        const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
    

    })
    
    test('debe mostrar la imagen con sus props', () => {

        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);

    })

    test('debe tener animate__heartBeat', () => {
        
        const div = wrapper.find('div');
        const className= div.prop('className');

        expect( className.includes('animate__heartBeat')).toBe(true);
    })
    
    
    
})