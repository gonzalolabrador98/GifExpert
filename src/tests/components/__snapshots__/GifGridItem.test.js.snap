// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem debe mostrar las pruebas correctamente 1`] = `
<div
  className="card animate__animated animate__heartBeat"
>
  <img
    alt="Un titulo"
    src="https://elquetepincha.com"
  />
  <p>
    Un titulo
  </p>
</div>
`;
